src/content/app/genome-browser/state/track-config/trackConfigSlice.ts

```diff
@@ -21,43 +21,73 @@ import {
   PayloadAction,
   ThunkAction
 } from '@reduxjs/toolkit';
+import { RootState } from 'src/store';
 
 import browserStorageService from 'src/content/app/genome-browser/services/browserStorageService';
+import browserTrackConfigStorageService from '../../components/browser-track-config/services/browser-track-config-storage-service';
+import { BrowserTrackStates } from 'src/content/app/genome-browser/components/track-panel/trackPanelConfig';
 
 import { updateTrackStates } from 'src/content/app/genome-browser/state/browser-general/browserGeneralSlice';
 import { getBrowserTrackStates } from 'src/content/app/genome-browser/state/browser-general/browserGeneralSelectors';
 
-import { BrowserTrackStates } from 'src/content/app/genome-browser/components/track-panel/trackPanelConfig';
-import { RootState } from 'src/store';
+import { GenomeTrackCategory } from 'src/shared/state/genome/genomeTypes';
 
 export type CogList = {
   [key: string]: number;
 };
 
-export type TrackConfigState = Readonly<{
+export enum TrackType {
+  GENE = 'gene',
+  REGULAR = 'regular'
+}
+
+export const getTrackType = (trackId: string) => {
+  if (trackId.startsWith('gene')) {
+    return TrackType.GENE;
+  } else {
+    return TrackType.REGULAR;
+  }
+};
+
+export type GeneTrackConfig = {
+  showMoreTranscripts: boolean;
+  showTranscriptIds: boolean;
+  showTrackName: boolean;
+  showFeatureLabel: boolean;
+  trackType: TrackType.GENE;
+};
+
+export type RegularTrackConfig = {
+  showTrackName: boolean;
+  trackType: TrackType.REGULAR;
+};
+
+export type TrackConfig = GeneTrackConfig | RegularTrackConfig;
+
+export type TrackInfo = {
+  [trackId: string]: TrackConfig;
+};
+
+export type TrackConfigStateForObject = Readonly<{
```

Comment:
I don't understand this name at all. What is it intended to mean? Does it refer only to tracks related to an "ens object" (in which case we are trying to call them entities, I believe)? The word "state" doesn't make the intent any clearer.

Reply:
This is not the ensobject, it is the configuration for a single track. any suggestions?

Reply:
Shouldn't this just be `TrackConfig`?

Reply:
Final types after discussion is below
```
type GenomeTrackConfigs = {
  [genomeId]: TrackConfigsPerGenome
}

type TrackConfigsPerGenome = {
  {
    applyToAllTracks: boolean
    tracks: TrackConfigs
  }
}

type TrackConfigs = {
  [trackId]: TrackConfig
}

type TrackConfig = {
  showTrackName: boolean
}```